lib/inets/src/http_lib/http_uri.erl

```diff
@@ -61,16 +61,30 @@
 	 scheme_defaults/0, 
 	 encode/1, decode/1]).
 
--export_type([scheme/0, default_scheme_port_number/0]).
+-export_type([uri/0,
+              user_info/0,
+              scheme/0, default_scheme_port_number/0,
+              host/0,
+              path/0,
+              query/0,
+              fragment/0]).
 
+-type uri() :: string() | unicode:unicode_binary().
```

Comment:
Using the type unicode:unicode_binary() "implies" that the module can properly handle UTF-8 encoded binaries but unfortunately it can't. I would prefer using 'binary()' instead.

Reply:
I understand this applies across the whole module. For some types - e.g. the ones that are output of functions - being proper UTF-8 binaries may apply but it looks too much effort ...

> As a side note, we are working on to remove application dependencies to the http_uri module and eventually obsolete it by the uri_string module in stdlib.

... especially considering the module is intended to be obsoleted.

Will propose change shortly.

Reply:
Relevant past comment https://github.com/erlang/otp/pull/1322#issuecomment-299409428

Reply:
Addressed in a31bdeb62a